Source/WebSocket/Websocketpp/websocketpp_websocket.cpp

```diff
@@ -528,7 +522,7 @@ struct wspp_websocket_impl : public hc_websocket_impl, public std::enable_shared
             auto pThis = reinterpret_cast<wspp_websocket_impl*>(async->context);
 
             // Wait for background thread to finish
-            (void)GetAsyncStatus(pThis->m_backgroundAsync, true);
+            pThis->m_websocketThread.join();
```

Comment:
Just looked at the Casablanca wspp implementation and they have this:

            if (m_thread.joinable())
            {
                m_thread.join();
            }